relm-examples/examples/7gui/5_crud/src/gui/person_list_box.rs

```diff
@@ -0,0 +1,128 @@
+    SelectionChanged(Option<usize>),
+}
+
+pub struct PersonListBoxModel {
+    persons: PersonList,
+    filtered_persons: PersonList,
+    filter: String,
+
+    selected: Option<Person>,
+
+    /// The stream for the window will be used to send signals to the window.
+    win_stream: StreamHandle<WinMsg>,
+}
+
+#[widget]
+impl Widget for PersonListBox {
+    fn model(_relm: &Relm<Self>, win_stream: StreamHandle<WinMsg>) -> PersonListBoxModel {
+        PersonListBoxModel {
+            persons: PersonList::new(),
+            filtered_persons: PersonList::new(),
+            filter: "".to_string(),
+
+            selected: None,
+
+            win_stream,
+        }
+    }
+
+    fn update(&mut self, event: PersonListBoxMsg) {
+        match event {
+            PersonListBoxMsg::AddPerson(person) => self.add_person(person),
+            PersonListBoxMsg::DeleteSelected => self.delete_selected(),
+            PersonListBoxMsg::UpdateSelected(person) => {
+                self.delete_selected();
+                self.add_person(person);
+            }
+            PersonListBoxMsg::Filter(filter) => {
+                self.model.filter = filter.clone();
+                self.update_filter();
+                self.update_list_box();
+            }
+            PersonListBoxMsg::SelectionChanged(index) => {
+                if let Some(idx) = index {
+                    let person = &self.model.filtered_persons[idx];
+                    self.model.selected = Some(person.clone());
+                } else {
+                    self.model.selected = None;
+                }
+
+                // Send the update to the parent widget.
+                self.model
+                    .win_stream
+                    .emit(WinMsg::UpdateSelected(self.model.selected.clone()));
+            }
+        }
+    }
+
+    view! {
+        #[name="list_box"]
+        gtk::ListBox {
+            selected_rows_changed(listbox) => {
+                let selected = listbox.get_selected_row();
+                if let Some(row) = selected {
+                    let index = listbox
+                        .get_children()
+                        .iter()
+                        .position(|r| r == &row)
+                        .unwrap();
+                    PersonListBoxMsg::SelectionChanged(Some(index))
+                } else {
+                    PersonListBoxMsg::SelectionChanged(None)
+                }
+            }
+        }
+    }
+}
+
+impl PersonListBox {
+    /// Updates the list box to represent the persons in `self.model.filtered_persons`.
+    fn update_list_box(&self) {
+        // Remove all rows from the list box.
+        let list_box = &self.widgets.list_box;
+        let list_box_clone = list_box.clone();
+        list_box.foreach(|c| list_box_clone.remove(c));
+
+        // Add all persons in the filter.
+        for person in self.model.filtered_persons.get_all() {
+            let _ = list_box.add_widget::<PersonRow>(person.clone());
```

Comment:
I made a custom Widget for `PersonRow` because it seemed sensible.
A relm widget is probably not neccessary (see definition of `PersonRow`, it has no logic just display).

Looking back at it, it would probably be better to insert the Label directly into the `ListBox`.